src/plugins/autologin/MQ2AutoLogin.cpp

```diff
@@ -429,19 +455,18 @@ void Cmd_Loginchar(SPAWNINFO* pChar, char* szLine)
 	auto record = ProfileRecord::FromString(szLine);
 	if (!record.profileName.empty() && !record.serverName.empty() && !record.characterName.empty())
 	{
-		record = ProfileRecord::FromINI(
-			record.profileName,
-			fmt::format("{}:{}_Blob", record.serverName, record.characterName),
-			INIFileName);
-
 		LoginProfile(
 			record.profileName.c_str(),
 			record.serverName.c_str(),
 			record.characterName.c_str());
 	}
-	else if (!record.serverName.empty() && !record.accountName.empty() && !record.accountPassword.empty())
+	else if (!record.accountName.empty() && !record.accountPassword.empty())
 	{
-		if (record.characterName.empty())
+		if (record.serverName.empty())
+			LoginServerSelect(
+				record.accountName.c_str(),
+				record.accountPassword.c_str());
+		else if (record.characterName.empty())
```

Comment:
more of a mental note, should probably convert all this code to use string_view or std::string later.